hudi-common/src/main/java/org/apache/hudi/common/config/HoodieMetadataConfig.java

```diff
@@ -554,6 +554,14 @@ public final class HoodieMetadataConfig extends HoodieConfig {
           + "bloom filter row for the files in the metadata table. Only applies if the filter "
           + "type (" + BLOOM_FILTER_TYPE.key() + " ) is BloomFilterTypeCode.DYNAMIC_V0.");
 
+  public static final ConfigProperty<Integer> WRITE_STATUS_COALESCE_PARALLELISM = ConfigProperty
+      .key(METADATA_PREFIX + ".write.status.coalesce.parallelism")
+      .defaultValue(0)
+      .markAdvanced()
+      .sinceVersion("1.1.0")
+      .withDocumentation("When set to a positive number, this config reduces the number of "
+          + "write status partitions to the set number when the current number of partitions is higher.");
```

Comment:
we can keep it generic. 
`when set to positive number, metadata table record preparation stages honor the set value for number of tasks. If not, number of write status's from data table writes will be used for metadata table record preparation`